dspy/primitives/base_module.py

```diff
@@ -233,26 +234,35 @@ def save(self, path, save_program=False, modules_to_serialize=None):
                     "with `.pkl`, or saving the whole program by setting `save_program=True`."
                 )
         elif path.suffix == ".pkl":
+            logger.warning("Loading .pkl files can run arbitrary code, which may be dangerous. Prefer "
```

Comment:
For the right action, let's include the code details for clear action, e.g., "Prefer saving with json format, e.g., module.save("module.json")"